eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -106,8 +114,8 @@ public function buildView(array $parameters)
             if ($location->contentId !== $content->id) {
                 throw new InvalidArgumentException('Location', 'Provided location does not belong to selected content');
             }
-        } else {
-            $location = $this->loadLocation($content->contentInfo->mainLocationId);
+        } else if (isset($this->locationLoader)) {
```

Comment:
`} else if (isset($this->locationLoader) && $content->getContentInfo()->mainLocationId) {`